src/utils/mergeTree.js

```diff
@@ -212,37 +233,53 @@ async function mergeBlobs({
   const type = 'blob'
   // Compute the new mode.
   // Since there are ONLY two valid blob modes ('100755' and '100644') it boils down to this
-  const mode =
-    (await base.mode()) === (await ours.mode())
-      ? await theirs.mode()
-      : await ours.mode()
-  // The trivial case: nothing to merge except maybe mode
-  if ((await ours.oid()) === (await theirs.oid())) {
-    return { mode, path, oid: await ours.oid(), type }
-  }
-  // if only one side made oid changes, return that side's oid
-  if ((await ours.oid()) === (await base.oid())) {
-    return { mode, path, oid: await theirs.oid(), type }
-  }
-  if ((await theirs.oid()) === (await base.oid())) {
-    return { mode, path, oid: await ours.oid(), type }
+  let mode = await ours.mode();
+  if (base !== null) {
+    const mode =
+      (await base.mode()) === (await ours.mode())
```

Comment:
do you need the `await ours.mode()` calls here? Can't it just be `mode`?

Reply:
good eye, this version is a rough 1st pass to prop everything up.  Thanks!

Reply:
I'd go further and say rip all of the `await`s out of the logic and up above into variables. I think combining the awaiting with the logic makes it harder to read than the pure logic.